File: app/src/main/java/com/example/testgithub/TestGithub.java
package com.example.testgithub;

import android.app.Activity;
import android.app.Application;
import android.content.Context;

import com.example.testgithub.dependencyinjection.AppInjector;
import com.facebook.stetho.Stetho;

import javax.inject.Inject;

import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;

public class TestGithub extends Application implements HasActivityInjector {

    @Inject
    DispatchingAndroidInjector<Activity> dispatchingAndroidInjector;


    @Override
    public void onCreate() {
        super.onCreate();

        Stetho.initializeWithDefaults(this);
        AppInjector.init(this);


    }

    @Override
    public DispatchingAndroidInjector<Activity> activityInjector() {
        return dispatchingAndroidInjector;
    }


}

File: app/src/main/java/com/example/testgithub/TestGithub.kt
package com.example.testgithub

import android.app.Activity
import android.app.Application

import com.example.testgithub.dependencyinjection.AppInjector
import com.facebook.stetho.Stetho

import javax.inject.Inject

import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector

class TestGithub : Application(), HasActivityInjector {

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Activity>


    override fun onCreate() {
        super.onCreate()

        Stetho.initializeWithDefaults(this)
        AppInjector.init(this)


    }

    override fun activityInjector() = dispatchingAndroidInjector

}
